hr_timesheet_role/models/project_project.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2018 Brainbean Apps (https://brainbeanapps.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models, api
+
+
+class ProjectProject(models.Model):
+    _inherit = 'project.project'
+
+    is_timesheet_role_required = fields.Boolean(
+        string='Timesheet Role Required',
+        default=lambda self: self._default_is_timesheet_role_required(),
+    )
+    limit_timesheet_role_to_assignments = fields.Boolean(
+        string='Limit Timesheet Role to Assignments',
+        default=(
+            lambda self: self._default_limit_timesheet_role_to_assignments()
+        ),
+    )
+
+    @api.model
+    def _default_is_timesheet_role_required(self):
+        company = self.env['res.company'].browse(
+            self._context.get('company_id', self.env.user.company_id.id)
+        )
+        return company.is_timesheet_role_required
+
+    @api.model
+    def _default_limit_timesheet_role_to_assignments(self):
+        company = self.env['res.company'].browse(
+            self._context.get('company_id', self.env.user.company_id.id)
+        )
+        return company.limit_timesheet_role_to_assignments
+
+    @api.model
+    def create(self, values):
+        company = None
```

Comment:
Is this not handled by the defaults of the fields itself?

Reply:
It's needed by the lines below, to set `limit_timesheet_role_to_assignments` - otherwise a non-UI created project will have wrong default values.

Reply:
(If you'll comment out these lines - tests will fail and show where exactly this is needed)